jarviscli/plugins/name_day.py

```diff
@@ -0,0 +1,222 @@
+
+    You can also see the name day of a specific name.
+    """
+    def __call__(self, jarvis, s):
+        self.jarvis = jarvis
+        self.main()
+
+    def __init__(self):
+        self.jarvis = None
+        self.location = ""
+        self.countries = {
+            "Austria": "at",
+            "Bulgaria": "bg",
+            "Croatia": "hr",
+            "Czech Republic": "cz",
+            "Denmark": "dk",
+            "Estonia": "ee",
+            "Finland": "fi",
+            "France": "fr",
+            "Germany": "de",
+            "Greece": "gr",
+            "Hungary": "hu",
+            "Italy": "it",
+            "Latvia": "lv",
+            "Lithuania": "lt",
+            "Poland": "pl",
+            "Russia": "ru",
+            "Slovakia": "sk",
+            "Spain": "es",
+            "Sweden": "se",
+            "United States": "us"
+        }
+
+    def main(self):
+
+        if self.location == "":
+            self.get_location()
+        while True:
+            # main menu
+            self.jarvis.say("It appears you are in " + self.location, color=Fore.BLUE)
+            self.jarvis.say("1 See Today's name days")
+            self.jarvis.say("2 See Tomorrow's name days")
+            self.jarvis.say("3 Chose specific date")
+            self.jarvis.say("4 Chose specific name")
+            self.jarvis.say("5 Chose an other country")
+            self.jarvis.say("6 Exit")
+
+            # get user input
+            try:
+                inp = int(self.jarvis.input("Select the desired number: ", color=Fore.GREEN))
+            except ValueError:
+                self.jarvis.say("Please select a number")
+                continue
+
+            # handle input
+            if inp == 1:
+                self.today()
+            elif inp == 2:
+                self.tomorrow()
+            elif inp == 3:
+                self.specific_date()
+            elif inp == 4:
+                self.specific_name()
+            elif inp == 5:
+                self.change_country()
+                continue
+            elif inp == 6:
+                break
+            else:
+                self.jarvis.say("Please select a valid number")
+
+            # ask user if he wants to continue
+            inp = self.jarvis.input("Do you want to continue? (Y/N)", color=Fore.RED)
+            if inp.lower() == "n":
+                break
+
+    def get_country_code(self):
+        """Return the two letter country code"""
+        return self.countries[self.location]
+
+    def get_location(self):
+        """
+        Get the location of the user.
+        If the user is in a supported country, the country is saved in location class variable.
+        otherwise, the user is asked to enter a country from the supported list.
+        """
+        self.jarvis.say("Getting Location ... ")
+        send_url = 'http://api.ipstack.com/check?access_key=f16ebe59174140a634827d674e605350&output=json&legacy=1'
```

Comment:
Don't really like api keys in source code...
I'll think of something - Jarvis dev branch has (or will have) support for getting location build directly into core.

I will come back to your PR later (and should I forget, comment something ;) )

Reply:
I took the code from here, but i genetated my own free API key
https://github.com/sukeesh/Jarvis/blob/2ed8cbd0655b4700da931962afd78747282b56dc/jarviscli/packages/mapps.py#L13-L15
The same API can be found here:
https://github.com/sukeesh/Jarvis/blob/700ea2e8b1f88aca5d5baf5df0103c9ec647fe15/jarviscli/plugins/location.py#L12

Notice that when i tried to run the two above code samples, the first one returned : `{'success': False, 'error': {'code': 104, 'type': 'usage_limit_reached', 'info': 'Your monthly usage limit has been reached. Please upgrade your Subscription Plan.'}}` and the second one worked fine.